src/kernel/include/shared/cos_config.h

```diff
@@ -32,6 +32,11 @@
 
 #define COS_MEM_KERN_VA_SZ (1 << 24) /* 16 MB from KERN_START_VA + end of kernel image onward */
 
+#define COS_HW_MMIO_START_VA 0xf0000000 /* Assuming all hardware virtual addresses are beyond this */
```

Comment:
I'll continue looking onwards, but this doesn't make much sense to me initially.  I'm likely missing the utility that I'll understand as I go on.  I'm guessing this is a reserved region to map in dynamically created physical apertures for MMIO?

Reply:
I commented on this. This is not a great way to fix the current issue. 
I think a better way is to have a limit on HW MMIO maps, so we can correctly set the addressable VA (kernel/user memory) memory.

It should really be `COS_HW_MMIO_MAP_MAX` and I'd set it to 64MB to start with.